### PR TITLE
added user profile section

### PR DESCRIPTION
**Pull request by RomanSaviuk**
dropdown in progress

### CODE REVIEW

File: client/screens/home.cpp [hunk 1]
```diff
@@ -4,9 +4,47 @@
...
+
+    QObject::connect(userProfileWidget, &userProfile::dropdownButtonClicked, this, &Home::onDropdownButtonClicked);
 }
 
 Home::~Home()
 {
     delete ui;
 }
+
+void Home::moveDropdown(){
```
Comment (AndriiM14) [inline client/screens/home.cpp:23]: Remove that feature, no need for that hack.

File: client/screens/user_profile/user_dropdown/user_dropdown.h [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+
+#include <QWidget>
+
+QT_BEGIN_NAMESPACE
+namespace Ui {
+class userDropdown;
+}
+QT_END_NAMESPACE
+
+class userDropdown : public QWidget
```
Comment (AndriiM14) [inline client/screens/user_profile/user_dropdown/user_dropdown.h:12]: Fix class naming (CamelCase)

File: client/screens/user_profile/user_login/user_login.cpp [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+    //connect(ui->dropdownButton, &QPushButton::clicked, this, &userProfile::onDropdownButtonClicked);
+}
+
+userLogin::~userLogin()
+{
+    delete ui;
+}
+//
+//void userLogin::onDropdownButtonClicked()
```
Comment (AndriiM14) [inline client/screens/user_profile/user_login/user_login.cpp:16]: remove that

File: client/screens/user_profile/user_login/user_login.h [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+
+#include <QWidget>
+
+QT_BEGIN_NAMESPACE
+namespace Ui {
+class userLogin;
+}
+QT_END_NAMESPACE
+
+class userLogin : public QWidget
```
Comment (AndriiM14) [inline client/screens/user_profile/user_login/user_login.h:12]: Same problems, comments and class naming

File: client/screens/user_profile/user_profile.cpp [hunk 5]
```diff
@@ -0,0 +1,19 @@
+#include "user_profile.h"
+#include "ui_user_profile.h"
+
+userProfile::userProfile(QWidget *parent) : QWidget(parent), ui(new Ui::userProfile)
```
Comment (AndriiM14) [inline client/screens/user_profile/user_profile.cpp:4]: u really forgot about that naming

File: client/screens/user_profile/user_profile.h [hunk 6]
```diff
@@ -0,0 +1,30 @@
...
+
+#include <QWidget>
+
+QT_BEGIN_NAMESPACE
+namespace Ui {
+class userProfile;
+}
+QT_END_NAMESPACE
+
+class userProfile : public QWidget
```
Comment (AndriiM14) [inline client/screens/user_profile/user_profile.h:12]: naming

File: client/screens/user_profile/user_dropdown/user_dropdown.cpp [hunk 7]
```diff
@@ -0,0 +1,22 @@
+#include "user_dropdown.h"
+#include "ui_user_dropdown.h"
+
+#include "QGraphicsDropShadowEffect"
+
+userDropdown::userDropdown(QWidget *parent) : QWidget(parent), ui(new Ui::userDropdown)
+{
+    ui->setupUi(this);
+
+    auto *effect = new QGraphicsDropShadowEffect;
```
Comment (AndriiM14) [inline client/screens/user_profile/user_dropdown/user_dropdown.cpp:10]: Memory leak - not deleting "effect" var, u allocated that on heap, bro.

File: client/screens/home.cpp [hunk 8]
```diff
@@ -4,9 +4,47 @@
 Home::Home(QWidget *parent) : QWidget(parent), ui(new Ui::Home)
 {
     ui->setupUi(this);
+
+    this->userProfileWidget = new userProfile(this);
```
Comment (AndriiM14) [inline client/screens/home.cpp:8]: Memory leak: new without delete.

**Review by AndriiM14 — CHANGES_REQUESTED** [review]

File: client/screens/user_profile/user_login/user_login.cpp [hunk 9]
```diff
@@ -0,0 +1,19 @@
+#include "user_login.h"
+#include "ui_user_login.h"
+
+userLogin::userLogin(QWidget *parent) : QWidget(parent), ui(new Ui::userLogin)
+{
+    ui->setupUi(this);
+
+    //connect(ui->dropdownButton, &QPushButton::clicked, this, &userProfile::onDropdownButtonClicked);
```
[the remark this answers is not in the document]
Comment (RomanSaviuk) [reply client/screens/user_profile/user_login/user_login.cpp:7]: we'll need it later

File: client/screens/home.cpp (hunk 8 above)
Comment (RomanSaviuk) [reply client/screens/home.cpp]: child is destroyed with parent

File: client/screens/user_profile/user_dropdown/user_dropdown.cpp (hunk 7 above)
Comment (RomanSaviuk) [reply client/screens/user_profile/user_dropdown/user_dropdown.cpp:10]: added parent argument, it'll deleted with dropdown

**Pull request merged**